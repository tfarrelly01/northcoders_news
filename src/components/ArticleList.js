import React from 'react';
import PropTypes from 'prop-types';
import ArticleCard from './ArticleCard';

const ArticleList = props => (
  <div>
    {props.articles.sort((a, b) => b.votes - a.votes)
      .map(article => {
        let userProfile = props.users.find(user => user.username === article.created_by);
        return (
          <ArticleCard
            key={article._id}
            id={article._id}
            title={article.title}
            createdBy={article.created_by}
            votes={article.votes}
            comments={article.comments}
            name={userProfile.name}
            avatarUrl={userProfile.avatar_url}
          />
        );
      })
    }
  </div>
);

ArticleList.defaultProps = {
  users: [],
  articles: []
};

ArticleList.propTypes = {
	users: PropTypes.array.isRequired,
	articles: PropTypes.array.isRequired
};

export default ArticleList;
