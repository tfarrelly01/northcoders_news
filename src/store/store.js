import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../reducer/reducer';

const logger = createLogger();

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk, logger),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
      ? window.devToolsExtension()
      : f => f
  )
);

export default store;