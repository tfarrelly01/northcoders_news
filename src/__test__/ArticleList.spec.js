import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../store/store';

import ArticleList from '../components/ArticleList';

describe('ArticleList Component', () => {
  test('ArticleList renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <ArticleList />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
