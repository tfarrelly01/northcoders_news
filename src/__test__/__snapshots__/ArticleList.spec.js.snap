// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Component ArticleList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ArticleList
    articles={Array []}
    users={Array []}
  />,
  "nodes": Array [
    <ArticleList
      articles={Array []}
      users={Array []}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <ArticleList
          articles={Array []}
          users={Array []}
        />
      </Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <ArticleList
            articles={Array []}
            users={Array []}
          />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ArticleList
          articles={Array []}
          users={Array []}
        />,
        "_debugID": 2,
        "_renderedOutput": <ArticleList
          articles={Array []}
          users={Array []}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ArticleList
      articles={Array []}
      users={Array []}
    />
  </Provider>,
}
`;
